fennel/testing/mock_client.py

```diff
@@ -518,6 +524,33 @@ def _get_secret(self, secret_name: str) -> Optional[str]:
         except KeyError:
             raise KeyError(f"Secret : `{secret_name}` does not exist.")
 
+    @staticmethod
+    def _resolve_output_features(
+        branch_class: Branch, outputs: List[Union[Feature, Featureset, str]]
+    ) -> List[Union[Feature, Featureset, str]]:
+        output_feature_names: List[Union[Feature, Featureset, str]] = []
+        for output in outputs:
+            if isinstance(output, Feature):
+                output_feature_names.append(output)
+            elif isinstance(output, Featureset):
+                output_feature_names.append(output)
+            elif isinstance(output, str):
+                try:
+                    is_valid_feature(output)
```

Comment:
why is is_valid_feature throwing an exception rather than returning a bool ?

Reply:
changed to bool.